request_test.go

```diff
@@ -772,6 +772,150 @@ func TestUnmarshalManyPayload(t *testing.T) {
 	}
 }
 
+func TestOnePayload_WithRelationLinks(t *testing.T) {
+	selfLink := "http://example.com/articles/1/relationships/author"
+	relatedLink := "http://example.com/articles/1/author"
+	sample := map[string]interface{}{
+		"data": map[string]interface{}{
+			"type": "posts",
+			"id":   "1",
+			"attributes": map[string]interface{}{
+				"body":  "First",
+				"title": "Post",
+			},
+			"relationships": map[string]interface{}{
+				"comments": map[string]interface{}{
+					"links": map[string]string{
+						"self":    selfLink,
+						"related": relatedLink,
+					},
+				},
+			},
+		},
+	}
+
+	data, err := json.Marshal(sample)
+	if err != nil {
+		t.Fatal(err)
+	}
+	in := bytes.NewReader(data)
+
+	out := new(Post)
+
+	if err := UnmarshalPayload(in, out); err != nil {
+		t.Fatal(err)
+	}
+
+	if len(out.Comments) == 0 {
+		t.Fatal("Was expecting non-empty Comments")
+	}
+
+	if out.Comments[0].Links == nil {
+		t.Fatal("Was expecting a non nil ptr Link field")
+	}
+
+	links := *out.Comments[0].Links
+	if links["self"] != selfLink {
+		t.Fatalf("Was expecting self Link to equal %s, but got %s", selfLink, links["self"])
+	}
+
+	if links["related"] != relatedLink {
+		t.Fatalf("Was expecting related Link to equal %s, but got %s", relatedLink, links["related"])
+	}
+}
+
+func TestOnePayload_WithLinks(t *testing.T) {
+	selfLink := "http://example.com/articles/1/relationships/author"
+	relatedLink := "http://example.com/articles/1/author"
+	sample := map[string]interface{}{
+		"data": map[string]interface{}{
+			"type": "comments",
+			"id":   "1",
+			"attributes": map[string]interface{}{
+				"body":  "First",
+				"title": "Post",
+			},
+			"links": map[string]string{
```

Comment:
Tests when `links` are at the same level as the other jsonapi tags. Relates to [this](https://github.com/hashicorp/jsonapi/pull/4/files#r605832120)